controller/api/auth.go

```diff
@@ -0,0 +1,78 @@
+package api
+
+import (
+	"net/http"
+	"os"
+	helper "web/helpers"
+	"web/model"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/render"
+)
+
+type loginFormData struct {
+	Email    string `form:"email"`
+	Username string `form:"username"`
+	Password string `form:"password"`
+}
+
+func Signup(c *gin.Context) {
```

Comment:
“Sign up” is a two-word verb phrase, so I would suggest naming the function like it, `SignUp`